packages/api-explorer/src/JsonForm/resolveReference.js

```diff
@@ -0,0 +1,9 @@
+import parser from '@apidevtools/json-schema-ref-parser'
+import {omit} from 'ramda'
+
+export default async function resolveReference (schema) {
+  const converted = await parser.dereference(schema)
+  // ref: https://github.com/APIDevTools/json-schema-ref-parser/issues/174
+  const convertedAgain = await parser.dereference(omit(['$ref'], converted))
```

Comment:
Why dereference again? The issue only talks about a ref at the root level (resolved by omit(['$ref'], converted))

Reply:
I have some doubts abouts the performance of the reference. Have you tested with a very big file? And with a lot of reference?

What about adding a memoize function around the resolve reference function?

Reply:
There is another bug that we have not open on that library.
If there is a "$ref" on the root, it does not derefence the reference nested. 


About the performance: this is done only at the mount of the component, and only on the "jsonSchema" passed to json-editor, not on the "whole" jsonSchema got from the backend

Reply:
Ok, could you add a comment about this?
But we render all json schema at the mount time, so it's on all the json schema from the backend.

Reply:
That could be a problem until we use `APIExplorer.forcePanelRender` prop, if we can remove it from our API Portal then there won't be a problem anymore, we should check whether it's still necessary

Reply:
done

Reply:
Wait, it's used for a reason. With a very large api spec, how fast it's rendered? We should test the performance, it was a big deal and we cannot returns to have an explorer slow

Reply:
We performed some performance recording with different setups, results follows.

If we remove `forcePanelRender` (in API Portal props) performance gets dramatically improved; but we have to make sure that we can safely remove it (as far as I remember it was added as a mitigation for the collapse opening bug).

If we keep it performance gets a little worse, we can perform some more benchmarks with other files.

| File                | Current Time | Time w/ Dereference | Current Time w/out forcePanelRender | Time w/ Dereference w/out forcePanelRender |
|---------------------|--------------|---------------------|-------------------------------------|--------------------------------------------|
| amazonaws.com:batch | ~3s          | ~3.7s               | ~400ms                              | ~200                                       |
| github.com          | > 1m         | > 1m                | ~1.2s                               | ~2s                                        |

Reply:
Some other (incomplete) benchmarks:


| File                | Current Time | Time w/ Dereference | Current Time w/out forcePanelRender | Time w/ Dereference w/out forcePanelRender |
|---------------------|--------------|---------------------|-------------------------------------|--------------------------------------------|
| amazonaws.com:batch | ~3s          | ~3.7s               | ~400ms                              | ~200                                       |
| github.com          | > 1m         | > 1m                | ~1.2s                               | ~2s                                        |
| multiple-securities | ~460ms       | ~335ms              |                                     |                                            |
| adafruit.com        | ~9s          | ~10s                |                                     |                                            |
| adobe.com:aem       | ~7.6s        | ~9s                 |                                     |                                            |
| bbc.com             | ~6.5s        |  ~5.6s              |                                     |                                            |
| clever-cloud.com    | > 1m         |                     |                                     |                                            |